build-system/tasks/helpers.js

```diff
@@ -827,31 +827,23 @@ function mkdirSync(path) {
 }
 
 /**
- * Returns the list of dependencies for a given JS entrypoint.
+ * Returns the list of dependencies for a given JS entrypoint by having esbuild
+ * generate a metafile for it. Uses the babel plugin that would've been used to
+ * build the entrypoint.
  * @param {string} entryPoint
+ * @param {!Object} options
  * @return {Promise<Array<string>>}
  */
-async function getDependencies(entryPoint) {
-  const stripImportAssertionPlugin = {
```

Comment:
Why were we only doing this previously?

Reply:
I'm not sure. From my reading of the code, we should be running all plugins that would've been run during compilation, and only then extract the list of dependencies from `metafile`. This will ensure that if we add new plugins in future to handle other kinds of special code, `getDependencies()` will automatically use them.

I added some debugging statements and verified that this works across a selection of JS entry points and extensions.

Reply:
The original PR description https://github.com/ampproject/amphtml/pull/33840#issue-615533919 says the following:

> Creates a new esbuild plugin for stripping import assertions. This will become unecessary once import assertions become stage 4 (and are built-in to esbuild)

Given that the watcher is broken right now, I think we should fix it by using whatever babel transforms we'd have used during compilation, and if required in future, add an additional plugin that does assertion stripping. We should run this by @samouri when he's back next week, but for now, the approach in this PR fixes compilation for JSX extensions.

@alanorozco Does this SGTY?

Reply:
Yep, thanks!

Reply:
Initially I was hoping to exclude babel from transpilations for performance reasons. I had assumed `esbuild` had jsx transforms built-in (which I still think it does, I just didn't enable them). 

Import assertions aren't builtin yet, so (but are handled by one of our babel transforms) so I created a simple string replace to handle it. @rsimha's logic in a separate thread that the TransformCache means there is a negligible cost of using our full babel pipeline is a great point that I hadn't considered!

tldr: This all LGTM